src/components/marker-chart/index.js

```diff
@@ -80,35 +87,36 @@ class MarkerChart extends React.PureComponent<Props> {
         </div>
         <MarkerChartCanvas
           key={threadIndex}
-          // ChartViewport props
-          isRowExpanded={isRowExpanded}
-          isSelected={isSelected}
-          timeRange={timeRange}
-          maxViewportHeight={maxViewportHeight}
-          maximumZoom={this.getMaximumZoom()}
-          selection={selection}
-          updateProfileSelection={updateProfileSelection}
-          viewportNeedsUpdate={viewportNeedsUpdate}
-          // MarkerChartCanvas props
-          rangeStart={timeRange.start}
-          rangeEnd={timeRange.end}
-          markerTimingRows={markerTimingRows}
-          maxMarkerRows={maxMarkerRows}
-          markers={markers}
-          rowHeight={ROW_HEIGHT}
+          viewportProps={{
+            timeRange: timeRange,
+            maxViewportHeight: maxViewportHeight,
+            maximumZoom: this.getMaximumZoom(),
+            selection: selection,
+            viewportNeedsUpdate: viewportNeedsUpdate,
```

Comment:
nit: you can use the ES6 shorthand object definition where possible: `{{ timeRange, maxViewportHeight, ... }}`